src/llama-model.cpp

```diff
@@ -550,6 +552,27 @@ void llama_model::load_hparams(llama_model_loader & ml) {
                     }
                 }
             } break;
+        case LLM_ARCH_LLAMA4:
+            {
+                ml.get_key(LLM_KV_ATTENTION_LAYERNORM_RMS_EPS, hparams.f_norm_rms_eps);
+                ml.get_key(LLM_KV_EXPERT_FEED_FORWARD_LENGTH,  hparams.n_ff_exp);
+                ml.get_key(LLM_KV_INTERLEAVE_MOE_LAYER_STEP,   hparams.n_moe_layer_step);
+                // hack: we use SWA to store the chunked attn mask
```

Comment:
Yes, SWA -> AUX makes sense.

Btw, we should soon implement actual SWA / chunked attention that uses less memory. It shouldn't be a big change and will improve memory usage significantly for such models.

Reply:
I'll do this after having the logits matching automated test. The problem is that changing name `n_swa` to `n_pattern_aux` make the code checking `if (n_swa) then using SWA` becomes invalid. It should now become `if (n_pattern_aux && is_swa) then use SWA`

I think it's better to add an enum called `llama_mask_aux` having 3 values: none, swa, chunked ; so that the code will become more clear